File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.py
class Solution:
    def maxDifference(self, s: str) -> int:
        ele = set(s)
        oddFreq = 0
        evenFreq = float('inf')
        for e in ele:
            if s.count(e) % 2 == 0:
                evenFreq = min(evenFreq, s.count(e))
            else:
                oddFreq = max(oddFreq, s.count(e))
        if not oddFreq or not evenFreq:
            return 0
        return oddFreq - evenFreq

File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
        HashMap<Character, Integer> ele = new HashMap<>();
        for (char c : s.toCharArray()) {
            ele.put(c, ele.getOrDefault(c, 0) + 1);
        }
        int oddFreq = 0;
        int evenFreq = Integer.MAX_VALUE;
        for(char e: ele.keySet()){
            if(ele.get(e) % 2 == 0){
                evenFreq = Math.min(evenFreq, ele.get(e));
            }else{
                oddFreq = Math.max(oddFreq, ele.get(e));
            }
        }
        return oddFreq - evenFreq;
    }
}